src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -52,6 +53,15 @@
     @Getter
     protected SoundInfo placeSound = new SoundInfo(Sound.BLOCK_WOOD_BREAK, 1F, 0.75F);
 
+    /**
+     * Determines whether or not the player is sneaking while placing a block.
+     * 
+     * @return If the player is sneaking while placing.
+     */
+    @Getter
+    @Setter
+    private boolean shiftClickPlace = false;
```

Comment:
This won't work properly in the case where multiple players try to interact with the same block in a short timespan. Instead, whatever method calls `canPlaceAt` or `canAttachTo` should check whether the *relevant* player `isSneaking()`.